homeassistant/components/thermostat/vera.py

```diff
@@ -0,0 +1,86 @@
+"""
+Support for Vera thermostats.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/switch.vera/
+"""
+import logging
+
+from homeassistant.util import convert
+from homeassistant.components.thermostat import (
+    STATE_AUTO, STATE_COOL, STATE_HEAT, STATE_OFF,
+    ThermostatDevice)
+from homeassistant.const import TEMP_FAHRENHEIT
+
+from homeassistant.components.vera import (
+    VeraDevice, VERA_DEVICES, VERA_CONTROLLER)
+
+DEPENDENCIES = ['vera']
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices_callback, discovery_info=None):
+    """Find and return Vera thermostats."""
+    add_devices_callback(
+        VeraThermostat(device, VERA_CONTROLLER) for
+        device in VERA_DEVICES['thermostat'])
+
+
+# pylint: disable=abstract-method
+class VeraThermostat(VeraDevice, ThermostatDevice):
+    """Representation of a Vera Thermostat."""
+
+    def __init__(self, vera_device, controller):
+        """Initialize the Vera device."""
+        VeraDevice.__init__(self, vera_device, controller)
+
+    @property
+    def current_power_mwh(self):
+        """Current power usage in mWh."""
+        power = self.vera_device.power
+        if power:
+            return convert(power, float, 0.0) * 1000
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return TEMP_FAHRENHEIT
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self.vera_device.get_current_temperature()
+
+    @property
+    def operation(self):
+        """Return current operation ie. heat, cool, idle."""
+        return self.vera_device.get_hvac_state()
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self.vera_device.get_current_goal_temperature()
```

Comment:
Does these methods make a call to the vera device? In that case they should be stored as instance variables inside the `update` method.